pkg/eth/api.go

```diff
@@ -889,74 +900,48 @@ func DoCall(ctx context.Context, b *Backend, args CallArgs, blockNrOrHash rpc.Bl
 			}
 		}
 	}
-	// Set default gas & gas price if none were set
-	gas := uint64(math.MaxUint64 / 2)
-	if args.Gas != nil {
-		gas = uint64(*args.Gas)
-	}
+	return nil
+}
 
-	if globalGasCap != nil && globalGasCap.Uint64() < gas {
-		logrus.Warnf("Caller gas above allowance, capping; requested: %d, cap: %d", gas, globalGasCap)
-		gas = globalGasCap.Uint64()
-	}
+// Call executes the given transaction on the state for the given block number.
+//
+// Additionally, the caller can specify a batch of contract for fields overriding.
+//
+// Note, this function doesn't make and changes in the state/blockchain and is
+// useful to execute and retrieve values.
+func (pea *PublicEthAPI) Call(ctx context.Context, args CallArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *StateOverride) (hexutil.Bytes, error) {
+	result, err := DoCall(ctx, pea.B, args, blockNrOrHash, overrides, 5*time.Second, pea.B.Config.RPCGasCap.Uint64())
 
-	var (
-		gasPrice  *big.Int
-		gasFeeCap *big.Int
-		gasTipCap *big.Int
-	)
+	// If the result contains a revert reason, try to unpack and return it.
+	if err == nil && len(result.Revert()) > 0 {
+		// Doubt: Should we return this? This might indicate failure in VM.
```

Comment:
Echoing what I said in slack: since it is not feasible to be certain whether or not the call was reverted due to incomplete state in our Postgres ETH-IPLD database or due to the nature of the call itself (that is, that it would fail on complete state), the safest thing to do is assume it was the former and not return the revert error and instead set the err var as it is being set currently such that we then call out to the proxy client to try and satisfy the request there.

If it is the former case, the proxy client will be able to satisfy the call (assuming it has the required state) and we will potentially fill in the missing data (with the call to `pea.writeStateDiffAtOrFor`, although that will only fill in the missing data if the missing nodes are in the statediff at this height/hash). If it is the later case, the proxy client will run into the same revert reason and the revert error will bubble up.